src/terms/ewald.jl

```diff
@@ -187,19 +186,24 @@ function energy_ewald(lattice, recip_lattice, charges, positions; η=nothing, fo
                     continue
                 end
 
-                dist = norm(lattice * (ti - tj - R))
+                v = lattice * (ti - tj - R)
+                dist = norm(v)
 
                 # erfc decays very quickly, so cut off at some point
                 if η * dist > max_erfc_arg
                     continue
                 end
 
                 any_term_contributes = true
-                sum_real += Zi * Zj * erfc(η * dist) / dist
+                energy_contribution = Zi * Zj * erfc(η * dist) / dist 
+                sum_real += energy_contribution
                 if forces !== nothing
-                    # Use ForwardDiff here because I'm lazy. TODO do it properly
-                    forces_real[i] -= ForwardDiff.gradient(r -> (dist=norm(lattice * (r - tj - R)); Zi * Zj * erfc(η * dist) / dist), ti)
-                    forces_real[j] -= ForwardDiff.gradient(r -> (dist=norm(lattice * (ti - r - R)); Zi * Zj * erfc(η * dist) / dist), tj)
+                    # grad = ForwardDiff.gradient(r -> (dist=norm(lattice * (r - tj - R)); Zi * Zj * erfc(η * dist) / dist), ti)
+                    ddist = Zi * Zj * η * (-2exp(-(η*dist)^2) / sqrt(T(π)))
```

Comment:
can you add a comment that ddist is the derivative of the energy wrt dist? (if that's what it is) I'd also name it dE_ddist or something like that.

Reply:
sounds good, I also renamed `dti` -> `dE_dti` accordingly